### ISSUE TITLE
RFE: enable logind "lingering" mode in an "transient" way, i.e. whose effect only lasts until reboot, and is forgotton afterwards

### ISSUE THREAD

**Issue by behrmann** [systemd/systemd#20682]
**Is your feature request related to a problem? Please describe.**

On a system with `KillUserProcesses=yes` when running something in a screen (or tmux) we have to `loginctl enable-linger` and then we can run them via `systemd-run --scope --user screen`. This way the `user@.service` is not killed once the last session of a lingering user closes, but lingering also starts the `user@.service` for that user on boot. At least, that is how I read Example 5 in the `systemd-run` man page and what it looks like when I try this out. 

Some middleground between a system with `KillUserProcesses=no` and `KillUserProcesses=yes` with lingering enabled would be nice; a system where user processes are killed by default and one can opt into not having certain processes killed without having a `user@.service` necessarily started on the next boot, i.e. a sort of transient lingering.

If I'm missing a straight-forward solution with the current state, than it would be great if this could be documented better.

**Describe the solution you'd like**

It would be great if a user could opt into a temporary, transient lingering when running something with  `systemd-run` or if a scope unit could opt into lingering, so that lingering is active only while there are running services in that scope. 

If I see it correctly, currently lingering is represented in the filesystem by empty files in `/var/lib/systemd/linger`. One way to implement transient lingering could be to make these files non-empty for transient lingering and have a service remove non-empty files before loginctl starts or loginctl remove those files itself.

**Describe alternatives you've considered**

One can run `loginctl enable-linger` when starting the screen/tmux and `loginctl disable-linger` before shutdown/when terminating the screen/tmux, but that's manual and might be forgotten.

**The systemd version you checked that didn't have the feature you are asking for**

> systemd 247 (247.3-6)

**Comment by poettering** [comment]
Not sure I fully grok what are you saying, but if we had `/run/systemd/linger/` instead of `/var/lib/systemd/linger/` then that's all you#d need? i.e. that you can mark users for lingering only until shutdown, but on next boot this is flushed out?

**Comment by behrmann** [comment]
Yes, having a `/run/systemd/linger/` would solve my usecase exactly.

**Comment by poettering** [comment]
hmm, is this really all you need? or do you also want a way to automatically enable such transient lingering on first login? i.e. do you also want a scheme how /run/systemd/linger/$UID is actually automatically created for you on first login?

**Comment by behrmann** [comment]
While this does sound neat at first, I think this goes a bit too much against `KillUserProcesses=yes`. I generally want everything of a user cleaned up upon logout, but as I see it, so that `systemd-run --scope --user screen` I need to enable lingering, which in turn starts the user service on boot. I'm happy about a solution that is fully opt-in, but only until reboot, since 
my usecase is that I would like to roll out `KillUserProcesses=yes` for all my users, but don't want them going around and enabling lingering on machines needlessly, just because they wanted to run a screen to keep looking at something. Since they will in general not need their user session running on that machine on the next boot.

Maybe `systemd-run` could grow a `--transient-linger`/`--linger-until-reboot` switch, to make this sort of usecase more seamless.
